src/solenoid/components.clj

```diff
@@ -41,38 +40,117 @@
+   (let [{:keys [id display-name value min max control-type]} control
+         {:keys [get-value-fn
+                 input-map-overrides
+                 value-container-map-overrides]} (merge base-component-default-opts opts)
+         form-key (control-type->form-key control-type)]
+     (intern 'user 'hx-vals (make-hx-vals control get-value-fn))
      [:div.control.row.my-1.mx-0.p-0
       [:span.col-3.text-end.mb-0.mt-2 (str (or display-name id))]
       [:span.col-7 {:class (str (name control-type) "-container")}
-       [:div.row.small
+       [:span.row.small
         (when min [:span.col-2.text-end.mb-0.mt-1 min])
-        [:input
+        [form-key
          (merge
            (make-input-map control)
-           {:hx-vals (make-hx-vals control get-value-js-str)}
+           {:hx-vals (make-hx-vals control get-value-fn)}
            control
-           input-map-overrides)]
+           input-map-overrides)
+         ;; if there's a form like a textarea, we need to insert the value within that tag so it shows up
+         (when (#{:textarea} form-key) (str value))]
         (when max [:span.col-2.mb-0.mt-1 max])]]
       [:span.col-2.mb-0.mt-1
        (merge
          {:id    (str (name id) "-value")
           :style {:display "none"}}
-         value-container-map-overrides) (str value)]])))
+         value-container-map-overrides) value]])))
 
-(defmulti render-controller type)
+(defmulti  render-controller type)
 (defmethod render-controller :default [m] (base-component m))
 (defmethod render-controller Num      [m] (base-component m))
 (defmethod render-controller Text     [m] (base-component m))
-(defmethod render-controller Slider   [m] (base-component m {} {:style {:display "inline-block"}}))
-(defmethod render-controller EdnBlock [m] (let [val (str (:value m))] (base-component (assoc m :value val))))
+(defmethod render-controller Toggle   [m]
+  (let [val (:value m)]
+    (base-component m {:input-map-overrides {:checked val}
+                       :get-value-fn (fn [control]
+                                       (str "document.getElementById('" (name (:id control)) "').checked"))})))
+
+(defmethod render-controller Slider [m]
+  (base-component m {:value-container-map-overrides
+                     {:style {:display "inline-block"}}}))
+
+(defmethod render-controller EdnBlock [m]
+  (let [val (str (:value m))]
+    (base-component (assoc m :value val))))
+
+(defn render-point-value
+  [m]
+  [:div {:style {:position "relative"
+                 :display  "flex"
+                 :left     (str (- (first (:value m)) 2) "px")
+                 :top      (str (- (second (:value m)) 2) "px")
+                 :margin-top "-2px"}}
+   [:div {:style {:width "4px" :height "4px" :border-radius "4px" :background "red"}}]
+   [:div {:style {:margin-top  "-0.45em !important"
+                  :margin-left "0.2em"
+                  :width       "auto"}} (str (:value m))]])
+
+(defmethod render-controller Point [control]
```

Comment:
Snuck in a whole new control type. As PRs go, this should not be put together with the 'derefs in letcontrols body' feature work, but it's still a bit of an unstructured project.

As I keep building out the project, this sort of thing will be more controlled and better separated.